### PR TITLE
K8s/Folders: Add frontend support for kubernetes create endpoint

### PR DESCRIPTION
**Pull request by suntala**
**What is this feature?**

Allows us to toggle which folder api endpoint the frontend should use when creating a folder.

**Why do we need this feature?**

The new create [endpoint](https://github.com/grafana/grafana/blob/main/pkg/api/api.go#L452) implemented by the k8s client will dual write folders to legacy and unified storage.

**Who is this feature for?**

Grafana Search and Storage team.

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/grafana-org/issues/210

**Special notes for your reviewer:**

Would be glad for some feedback from frontend folks. While it seems to be doing what we (search and storage team) need it to, I'm sure you have thoughts on how to better structure the change. 

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

File: public/app/features/manage-dashboards/state/actions.ts [hunk 1]
```diff
@@ -281,7 +282,7 @@ export async function moveFolders(folderUIDs: string[], toFolder: FolderInfo) {
 }
 
 export function createFolder(payload: any) {
-  return getBackendSrv().post('/api/folders', payload);
+  return getFolderAPI().createFolder(payload);
 }
```
Comment (joshhunt) [inline public/app/features/manage-dashboards/state/actions.ts:286]: The only place this function is used is in Alerting, and it actually should be deleted.

The rest of the frontend all makes their API requests through an RTK Query slice to properly handle caching and invalidation https://github.com/grafana/grafana/blob/joshhunt/deprecate-create-folder-fn/public/app/features/browse-dashboards/api/browseDashboardsAPI.ts#L112-L135

We'll need to come up with a way to swap out the API... which is much more difficult when the APIs accept and return differently shaped data...

Comment (suntala) [reply public/app/features/manage-dashboards/state/actions.ts:286]: > The only place this function is used is in Alerting, and it actually should be deleted.

Good to know. I am able to make calls to the new endpoint and create a folder that way without any issue though so I am not sure how that's happening.

Comment (joshhunt) [reply public/app/features/manage-dashboards/state/actions.ts:286]: From where in the UI? 🤔

Comment (suntala) [reply public/app/features/manage-dashboards/state/actions.ts:286]: If I go to `/dashboards` and then click on the "New" button > "create folder"

Comment (joshhunt) [reply public/app/features/manage-dashboards/state/actions.ts:286]: Hmmm interesting. Are you sure that's using the new K8s API? I just tested it by changing `createFolder` to throw an error:

```
export function createFolder(payload: any) {
  throw new Error("using deprecated function 'createFolder'");
  return getBackendSrv().post('/api/folders', payload);
}
```

and the folder was created, indicating this method is not called there. Looking at the code, I'm pretty sure it's using a different code path  (calling the RTK Query methods instead) https://github.com/grafana/grafana/blob/63841fefd64bb7bdd1d5d80a802c185ce6fa6153/public/app/features/browse-dashboards/components/CreateNewButton.tsx/#L27

Comment (suntala) [reply public/app/features/manage-dashboards/state/actions.ts:286]: I'll double check, but I think so. Depending on whether I enable `kubernetesFolders` or not, the payload changes. For example, we haven't populated the "url" field yet and it doesn't appear when the feature is enabled.

How do I include a new API endpoint in the RTK Query methods? I searched for `api/folders` and only found references to the other endpoints.

Comment (suntala) [reply public/app/features/manage-dashboards/state/actions.ts:286]: You're right. That's what it was. Now I need to reevaluate the changes this PR was trying to make. Moving this back to draft for now. Thank you very much for the input!

**Comment by suntala** [comment]
Need to reevaluate the changes this PR was trying to make. Closing for now.

**Pull request closed without merging**